### PR TITLE
Process multiple records

### PR DESCRIPTION
**Pull request by phamann**
One thing the load test showed yesterday is that the end-to-end process is quite slow (took ~40mins). After some research I realised this is because the AWS Kinesis -> Lambda event source can only process one function concurrently per shard. Therefore you either need to up the amount of Kinesis shards or pull more records of the queue each invocation. I've opted for the latter.

Sadly this doesn't make the function as pure as it was, but should give a massive perf improvement.

I'm going to try this with 10 messages at a time to see what impact it has on the overall time.

*From the AWS docs*:
> For Lambda functions reading off Amazon Kinesis and DynamoDB streams, Lambda executes your function concurrently for each shard on the stream. For example, if your Amazon Kinesis stream has 100 shards, your function will have at most 100 concurrent executions at any given time. The function reads serially off each active shard multiple times per second, invoking your function with as many records as are available, up to the batch size you have configured as part of your event source definition. For example, suppose you have a Amazon Kinesis stream with 10 active shards. In this scenario, your Lambda function will have 10 concurrent executions. For more information about the pull model, see The Pull Event Model.

http://docs.aws.amazon.com/lambda/latest/dg/intro-core-components.html#concurrent-executions

### CODE REVIEW

File: functions/create-email/index.js [hunk 1]
```diff
@@ -38,8 +34,19 @@ exports.handle = λ(event => {
...
 				return Promise.resolve();
 			});
+}
+
+exports.handle = λ(event => {
+	let users;
+
+	if(Array.isArray(event.Records)) {
+		users = event.Records.map(parseKinesisRecord);
+	} else {
```
Comment (keirog) [inline functions/create-email/index.js]: When might the else block get called?

File: functions/create-email/index.js [hunk 2]
```diff
@@ -38,8 +34,19 @@ exports.handle = λ(event => {
...
+		users = [event.uuid];
```
Comment (keirog) [inline functions/create-email/index.js]: Why the difference here?

Comment (phamann) [inline functions/create-email/index.js]: Good question:
- When we want to test locally and not have to fake a buffered kinesis stream.
- When we want to single requests via API gateway in the future for HTML rendering

**Comment by keirog** [comment]
👍

**Pull request merged**